cli/tests/pcluster/validators/test_iam_validators.py

```diff
@@ -113,3 +115,125 @@ def test_additional_iam_policy_validator(mocker, policy_arn, expected_get_policy
     mocker.patch("pcluster.aws.iam.IamClient.get_policy", side_effect=expected_get_policy_side_effect)
     actual_failures = AdditionalIamPolicyValidator().execute(policy=policy_arn)
     assert_failure_messages(actual_failures, expected_message)
+
+
+@pytest.mark.parametrize(
+    "resource_prefix, expected_message,expected_failure_level",
+    [
+        (
+            r"\pathprefix\\",
+            "Unsupported format for ResourcePrefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "",
+            "Unsupported format for ResourcePrefix ",
+            FailureLevel.ERROR,
+        ),
+        (
+            "/",
+            "Unsupported format for ResourcePrefix /",
+            FailureLevel.ERROR,
+        ),
+        (
+            "//",
+            "Unsupported format for ResourcePrefix //",
+            FailureLevel.ERROR,
+        ),
+        (
+            "///////",
+            "Unsupported format for ResourcePrefix ///////",
+            FailureLevel.ERROR,
+        ),
+        (
+            "/prefix",  # This is not pathprefix ---- Need to check
+            "Unsupported format for ResourcePrefix /prefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            ",./pathprefix/",
+            "Unsupported format for ResourcePrefix ,./pathprefix/",
+            FailureLevel.ERROR,
+        ),
+        (
+            ";)*/anything/pathprefix/",
+            "Unsupported format for ResourcePrefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "#$/pathprefix/anything/",
+            "Unsupported format for ResourcePrefix ",
+            FailureLevel.ERROR,
+        ),
+        (
+            "&&&roleprefix",
+            "Unsupported format for ResourcePrefix &&&roleprefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "@@/pathprefix/roleprefix",
+            "Unsupported format for ResourcePrefix @@/pathprefix/roleprefix",
+            FailureLevel.ERROR,
+        ),
+        (
+            "+/anything/pathprefix/roleprefix",
+            "Unsupported format for ResourcePrefix ",
+            FailureLevel.ERROR,
+        ),
+        (
+            "=/pathprefix/anything/roleprefix",
+            "Unsupported format for ResourcePrefix =/pathprefix/anything/roleprefix",
+            FailureLevel.ERROR,
+        ),
+        ("/prefix/", None, None),
+        ("/somepath/pathprefix/roleprefix", None, None),
+        ("/prefix/", None, None),
+        ("roleprefix", None, None),
+        ("/pathprefix/parallelcluster/", None, None),
+        ("/parallelcluster/pathprefix/", None, None),
+        ("/pathprefix/", None, None),
+        ("role-prefix", None, None),
```

Comment:
We can reduce the number of cases to the significative ones.
It would make sense to have one case including all the special characters that are accepted.
For example: `role-prefix@-+=._,`

Reply:
Sure